### PR TITLE
fix(jqLite.attr): ignore comment, text and attribute nodes when setting/...

### LINKED ISSUE #11038: JQLite issue setting attributes on comments ¯\_(ツ)_/¯
Here's [a demo](http://jsbin.com/rojexo/2/edit) of the issue (open the console) and you'll see something like this (asside from all the jsbin logs):

screen shot 2015-02-11 at 5 31 22 pm

This is referencing [this line](https://github.com/angular/angular.js/blob/d5b1d6edde194da75f235349783a0712aaab5f7e/src/jqLite.js#L607) which chokes if the element is a comment (evidently). jQuery proper handles this (somehow). If you edit the jsbin, simply uncomment the script tag to include jQuery and you'll see that the error is now gone.

I apologize if the demo is a little contrived, it was how I discovered the bug.

### PR DESCRIPTION
**Pull request by RobiFerentz**
...getting attributes

follow jQuery handling of the attr function

Fixes #11038

### CODE REVIEW

File: src/jqLite.js [hunk 1]
```diff
@@ -587,6 +587,9 @@ forEach({
   },
 
   attr: function(element, name, value) {
+    if (element.nodeType === 2 || element.nodeType === 3 || element.nodeType === 8) {
```
Comment (lgalfaso) [inline src/jqLite.js]: * please use `NODE_TYPE_TEXT` and `NODE_TYPE_COMMENT`, for nodeType === 2, please create `NODE_TYPE_ATTRIBUTE` to src/Angular.js and use it
* add `var nodeType = element.nodeType` and do the comparison to it

File: test/jqLiteSpec.js [hunk 2]
```diff
@@ -612,6 +612,14 @@ describe('jqLite', function() {
       expect(elm.attr('readOnly')).toBeUndefined();
       expect(elm.attr('disabled')).toBeUndefined();
     });
+
+    it('should ignore nodes that cannot have attributes', function() {
```
Comment (lgalfaso) [inline test/jqLiteSpec.js]: should do nothing when trying to set an attribute to a comment text node or attribute node

and please add the tests for all of them

**Comment by lgalfaso** [comment]
otherwise, LGTM

File: src/jqLite.js (hunk 1 above)
Comment (RobiFerentz) [inline src/jqLite.js]: I've added the NODE_TYPE_ATTRIBUTE = 2 to src/Angular.js (also in the jshint comment at the top, to expose it as global). However, running grunt test still gives me an error that NODE_TYPE_ATTRIBUTE is not defined. I must be missing something.

**Comment by RobiFerentz** [comment]
Done. :satisfied:

**Comment by lgalfaso** [comment]
landed as bb5bf7f8162d11610a53428e630b47030bdc38e5

**Pull request closed without merging**